### ISSUE TITLE
Uncaught TypeError: mergedOptions.onMissingTranslation is not a function

### ISSUE THREAD

**Issue by rohan-buchner** [ryandrewjohnson/react-localize-redux#103]
I'm not sure if this is a bug or a feature, as its kind of jumping outside of the bounds of the documentation.

I currently have a boot-server file where I follow [Option 2](https://github.com/ryandrewjohnson/react-localize-redux/issues/96) from this thread using the exposed action creators, as I inject the localisations once off during server render. (this was how I injected the localisations in the prior version)

   ```
export default createServerRenderer((params) => {
    return new Promise<RenderResult>((resolve, reject) => {

     // Prepare Redux store with in-memory history, 
     const store = configureStore(createMemoryHistory());

     const { translations } = params.data;

     const onMissingTranslation = ({ translationId, languageCode }) => {
            console.log(`Missing translationId: ${ translationId } for language: ${ languageCode }`);
            return `Label Not Set`;
        };

        // get translation files
        store.dispatch(initialize({
            languages: [{ name: 'English', code: 'en' }],
            translation: translations,
            options: {
                onMissingTranslation,
                renderInnerHtml: true,
                renderToStaticMarkup,
                defaultLanguage: 'en',
            }
        }));
        store.dispatch(addTranslationForLanguage(translations, 'en'));
        store.dispatch(setActiveLanguage('en'));

       // some code   

        const app = (
            <LocalizeProvider store={store}>
                <Provider store={store}>
                    <StaticRouter context={routerContext} location={params.location.path}
                                  children={routes(instanceType, appRoutes)}/>
                </Provider>
            </LocalizeProvider>
        );    

        // etc...
    });
});
```
It works for the most part.
That being said... since swapping over, and using the technique above. I have not been able to get the onMissingTranslations to actually work (javascript just explodes if there is no translation for a key) keep getting this error 'Uncaught TypeError: mergedOptions.onMissingTranslation is not a function' even with the above set.

Even when not setting the onMissingTranslation, I get the same issue... thus leading me to believe my setup or boot code might be somewhat wonky.

Versions:
 "react": "16.2.0",
 "react-redux": "5.0.6",
 "react-localize-redux": "3.1.1",

Side note... I'm not able to upgrade react version yet, as they're aiming to remove some hooks that I depend on for SSR purposes... so I'm giving it some time until they stabilise.

**Comment by thchia** [comment]
@rohan-buchner I am guessing that the `app` in your code snippet is eventually rendered using one of the `ReactDOM/server` render methods? (e.g. `renderToString()`)

If so my initial guess is that the store is not subscribed to properly, because this subscription happens in `componentDidMount()` which is not called on server render.

If possible can you create a minimal repro on CodeSandbox - it may be as simple as calling your `createServerRenderer()` to get the error to show itself.

**Comment by rohan-buchner** [comment]
@thchia thats correct.

Erm its somewhat difficult for me to create a minimal repo as i have a dotnetcore based project so my ssr is handled via that (internally is still uses node... but its a black box to me)... but i will see if i can maybe get something bashed out in a mutual language we both can work/debug from.

I 100% agree with your first guess actually... componentDidMount does not get called during ssr... thus I'd say its a safe bet that that is in fact the issue... 

Only thing is... is this something that will/should be addressed, as I'm assuming its not really a concern for this project?

**Comment by thchia** [comment]
Let me try and reproduce this to confirm the issue (though can’t promise on speed as I’m quite busy).

If it is indeed what I guessed above then we can consider it a bug and it is definitely something that should be addressed 🙂.

**Comment by ryandrewjohnson** [comment]
@rohan-buchner I've never done server side rendering on my end with the library, so I'm hoping you can help me out with some testing. 

This is the snippet of code that is currently called from `componentDidMount` in `LocalizeProvider`. 

```javascript
componentDidMount() {
    if (this.props.store) {
      this.unsubscribeFromStore = storeDidChange(
        this.props.store,
        this.onStateDidChange.bind(this)
      );
    }
  }
```
Locally in your app can you edit the `LocalizeProvider` code and insert this under the existing logic in the constructor, and remove the `componentDidMount` version. The constructor should be called server side as well.

The best case would still be if you could provide a codesandbox, but if that is not doable the above would help out a lot.

**Comment by samuelg0rd0n** [comment]
@ryandrewjohnson I'll be happy to help you. I'm just not sure what exactly you mean by editing the `LocalizeProvider` code. I have a react-localize-redux package in node_modules. What file exactly should I modify? lib/LocalizeProvider.js?

**Comment by rohan-buchner** [comment]
@ryandrewjohnson whoa... I seem to have disappeared in the void... we had some flu going round at the office... let me take a stab at that and let you know whats up. sorry for the delay in my response. thanks for the snippet

**Comment by ryandrewjohnson** [comment]
Fixed in #127  and included in version v3.4.0.
